lib/hl7/message.ex

```diff
@@ -131,6 +131,20 @@ defmodule HL7.Message do
     msg
   end
 
+  def new(<<"MSH|^~\\&", _rest::binary()>> = raw_text) do
```

Comment:
Is it okay to assume the input is using the standard delimiters?

Reply:
yup :) The parser code is only used for default limiters (as with this case). I found that I lost the performance benefits if I made them dynamic and used pinned case stuff to split based on the separators in the message. I've never seen an HL7 message that didn't use the defaults, though, so having this high performance path seems like a good way to go.

Reply:
If the separators are not the defaults, it's still using the old code.